### PR TITLE
2285 add sensors link under cold chain nav and sensor page with a list view of sensor information

### LINKED ISSUE #2285: Add `Sensors` link under ColdChain nav and sensor page with a list view of sensor information
## Describe the solution you'd like 🎁
- Add `Sensors` sub-link under `Cold Chain menu`
- Clicking on link leads to the sensors list view
- Hook to query sensor information
- Columns: 
   - Name
   - Storage Location
   - CCE
   - Serial Number
   - Battery Level
   - Last reading
   - Sensor Type
   - Breach
   - Last recording date / time
- Clicking on a row will direct user to sensor detail view  

Mockup: 
Screenshot 2023-09-26 at 12 18 46 PM

### Estimate
3 hours

### Moneyworks Jobcode 🧰
OMS:COLC

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #2285 

# 👩🏻‍💻 What does this PR do? 
This PR sets up the frontend for `Cold chain` and also contains the List View for cold chain. Have also added latest log and breach to Sensor Node to load in the `Sensors List View`
Screenshot 2023-10-10 at 15 21 40

# 🧪 How has/should this change been tested? 
- [ ] Have a cold chain data file
- [ ] Must have `Vaccine module` preference turned on
- [ ] Click on Sensor
- [ ] See a list view of all sensors

### CODE REVIEW

File: client/packages/coldchain/src/Sensor/ListView/ListView.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  useUrlQueryParams,
+  Formatter,
+} from '@openmsupply-client/common';
+import { useSensor, SensorFragment } from '../api';
+
+export const SensorListView: FC = () => {
+  const {
+    updateSortQuery,
+    updatePaginationQuery,
+    // filter,
```
Comment (roxy-dao) [inline client/packages/coldchain/src/Sensor/ListView/ListView.tsx:19]: Will implement once filter issue is done

File: server/graphql/types/src/types/sensor.rs [hunk 2]
```diff
@@ -70,6 +81,79 @@ impl SensorNode {
...
+    }
+
+    pub async fn latest_temperature_log(
+        &self,
+        ctx: &Context<'_>,
+    ) -> Result<Option<TemperatureLogConnector>> {
+        let filter = TemperatureLogFilter::new()
+            .sensor(SensorFilter::new().id(EqualFilter::equal_to(&self.row().id)));
+
+        let latest_log = get_temperature_logs(
```
Comment (roxy-dao) [inline server/graphql/types/src/types/sensor.rs:120]: Only needed the one entry, so didn't use loader... same as breach below

File: client/packages/common/src/intl/locales/en/common.json [hunk 3]
```diff
@@ -31,6 +31,10 @@
   "button.select-a-color": "Select a colour",
   "button.try-again": "Try again",
   "button.view": "View",
+  "cold-chain.cold-consecutive": "Consecutive",
```
Comment (roxy-dao) [inline client/packages/common/src/intl/locales/en/common.json:34]: Not sure if we want to say hot/cold with this since some mocks have it but some don't

**Review by mark-prins — COMMENTED** [review]
when running on postgres, there's an additional migration step which has been missed in the feature branch. I also found it odd to have the store preference included in the sensor migration file, so I've created a new `permissions_preferences` migration:

```
use crate::{migrations::sql, StorageConnection};

#[cfg(feature = "postgres")]
pub(crate) fn migrate(connection: &StorageConnection) -> anyhow::Result<()> {
    use crate::migrations::sql;

    sql!(
        connection,
        r#"ALTER TABLE store_preference ADD COLUMN vaccine_module bool NOT NULL DEFAULT false;
        ALTER TYPE permission_type ADD VALUE 'SENSOR_QUERY';
        ALTER TYPE permission_type ADD VALUE 'SENSOR_MUTATE';
        "#
    )?;

    Ok(())
}

#[cfg(not(feature = "postgres"))]
pub(crate) fn migrate(_connection: &StorageConnection) -> anyhow::Result<()> {
    sql!(
        connection,
        r#"
            ALTER TABLE store_preference ADD COLUMN vaccine_module bool NOT NULL DEFAULT false;
        "#
    )?;

    Ok(())
}
```

**Review by mark-prins — APPROVED** [review]
👍 great to see the cold chain come alive!

I had updated the icons and stories, so pushed those changes into a [suggestion branch](https://github.com/openmsupply/open-msupply/compare/2285-Add-Sensors-link-under-ColdChain-nav-and-sensor-page-with-a-list-view-of-sensor-information...2285-suggestion) just in case that's useful. 

The main thing in the icon changes is that the `ListIcon` file was incorrectly named, and some icons weren't in the story.

The other change is moving around the columns, so that the 'last reading' is beside the datetime, which is related directly to it.

File: client/packages/common/src/ui/icons/Thermometer.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import SvgIcon, { SvgIconProps } from '@mui/material/SvgIcon';
+
+export const ThermometerIcon = ({ ...props }: SvgIconProps): JSX.Element => {
```
[the remark this answers is not in the document]
Comment (roxy-dao) [reply client/packages/common/src/ui/icons/Thermometer.tsx:4]: I think it looks better with the markings on the side! Thanks Mark (:

File: client/packages/common/src/intl/locales/en/common.json (hunk 3 above)
Comment (roxy-dao) [reply client/packages/common/src/intl/locales/en/common.json:34]: Maybe I'll add the `hot/cold` since you can't really tell in sensor list view 🤔

**Comment by adamdewey** [comment]
Is there a way we can indicate that the temperature reading is in Celsius? 

Be good to remove any doubt..

**Comment by roxy-dao** [comment]
> Is there a way we can indicate that the temperature reading is in Celsius?
> 
> Be good to remove any doubt..

Are all the countries using celsius?

**Comment by adamdewey** [comment]
> > Is there a way we can indicate that the temperature reading is in Celsius?
> > Be good to remove any doubt..
> 
> Are all the countries using celsius?

I imagine so - I think Celsius would be the default for sure for pretty much everyone except the US

**Pull request merged**